pkg/codegen/internal/test/testdata/external-resource-schema/dotnet/ArgFunction.cs

```diff
@@ -13,6 +13,18 @@ public static class ArgFunction
     {
         public static Task<ArgFunctionResult> InvokeAsync(ArgFunctionArgs? args = null, InvokeOptions? options = null)
             => Pulumi.Deployment.Instance.InvokeAsync<ArgFunctionResult>("example::argFunction", args ?? new ArgFunctionArgs(), options.WithVersion());
+
+        public static Output<ArgFunctionResult> Invoke(ArgFunctionOutputArgs? args = null, InvokeOptions? options = null)
+        {
+            args = args ?? new ArgFunctionOutputArgs();
+            return Pulumi.Output.All(
+                args.Name.Box()
+            ).Apply(a => {
+                    var args = new ArgFunctionArgs();
+                    a[0].Set(args, nameof(args.Name));
```

Comment:
The Box business is somewhat messy but that's what I ended up with to workaround the problem that (1) tuple helpers `Output[A] * Output[B] -> Output[A*B]` only work up to 8, and mixing them up gets ugly really fast, while the `All` combinator used here requires all the arguments to have the same type. In addition, the `a[0].Set()` pattern uses reflection as I didn't find a way that C# type system understands to map this back to the properties without double-stating the type of the property  in the code (ouch). I've been really really wishing this was in F# when writing this..

Now I was somewhat hesitant but if this is a problem for reviewers I can rework this, now that the Go codegen test actually *runs unit tests* against the generated code, I've got a higher confidence in iterating on this.

Reply:
Seems reasonable to me.